tsl/src/nodes/decompress_chunk/decompress_chunk.c

```diff
@@ -210,6 +212,20 @@ build_compressed_scan_pathkeys(SortInfo *sort_info, PlannerInfo *root, List *chu
 
 			sortop =
 				get_opfamily_member(pk->pk_opfamily, var->vartype, var->vartype, pk->pk_strategy);
+			if (!get_ordering_op_properties(sortop, &opfamily, &opcintype, &strategy))
+			{
+				if (type_is_enum(var->vartype))
+				{
+					sortop = get_opfamily_member(pk->pk_opfamily,
+												 ANYENUMOID,
+												 ANYENUMOID,
+												 pk->pk_strategy);
+				}
+				else
+				{
+					elog(ERROR, "Invalid segmentby column");
+				}
```

Comment:
Do we need this branch? If yes shouldn't have a test case for this?

Reply:
Agree that its good to have a testcase. However iam not sure what test will hit this code.
I think it's better to have this code, so that it may hit in future. If we remove this, then code may further execute and report some unknown error or even worse case crash or return wrong results.